ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/ExecutionEnvironmentSection.java

```diff
@@ -270,16 +285,19 @@ public void run() {
 	}
 
 	private void updateButtons() {
+		String javaExecutionEnvironmentMatcher = "(?i)\\b(jre|javase|j2se).*\\b"; //$NON-NLS-1$
```

Comment:
To improve the performance you should in general consider to compile the regex into a static final `java.util.regex.Pattern`.
But I wonder if a simple `text.startswith("jre") || text.startswith("javase") || text.startswith("j2se")` would be simpler to read (and faster).